ims/client.go

```diff
@@ -76,7 +76,8 @@ type Response struct {
 	// The raw body of the HTTP response.
 	Body []byte
 	// The value of the X-Debug-Id header.
-	XDebugID string
+	XDebugID   string
+	RetryAfter string
```

Comment:
food for though, you looked more into the usage of retry after.
from my view it makes sense extract the returned time and parse it into a date-time type directly in the client, instead of delegating this responsibility to the caller.